xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3132,8 +3144,52 @@ int CVideoPlayer::GetSubtitle()
   return m_SelectionStreams.IndexOf(STREAM_SUBTITLE, *this);
 }
 
+void CVideoPlayer::UpdateStreamInfos()
+{
+  CSingleLock lock(m_player_status_lock);
+  int streamId;
+  std::string retVal;
+
+  // video
+  streamId = GetVideoStream();
```

Comment:
The class SelectionStreams is utter crap and needs re-write anyways. For now you can either expose its lock or add methods to grab/release the lock